### PR TITLE
Adds 'popper scaffold' test 

### PR DESCRIPTION
**Pull request by vipulchhabra99**
The popper scaffold test has been updated using click unit testing.

### CODE REVIEW

File: cli/test/test_scaffold.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import os
+
+
+class TestScaffold(PopperTest):
+
+    def test_scaffold(self):
+
+        repo = self.mk_repo()
+        runner = CliRunner()
+        file_loc = repo.working_dir+'/wf.yml'
```
Comment (ivotron) [inline cli/test/test_scaffold.py:17]: `mk_repo` creates a git repo. In this case, we don't really need one, so we can instead use `tempfile`:

```python
wf_dir = tempfile.mkdtemp()
file_loc = f'{wf_dir}/wf.yml'
```

BTW, we can use f-strings here as well as shown above, in order to keep it consistent with the rest of the codebase

File: cli/test/test_scaffold.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+        result = runner.invoke(scaffold.cli, ['-f', file_loc])
+
+        self.assertEqual(result.exit_code, 0)
+        self.assertTrue(os.path.isfile(file_loc))
```
Comment (ivotron) [inline cli/test/test_scaffold.py:22]: we can also check the content of the file to ensure that scaffolding did what it should. To do this, we can create a `Workflow` object from the file, and then compare it as shown [here](https://github.com/systemslab/popper/blob/master/cli/test/test_parser.py#L246).

File: cli/test/test_scaffold.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+        with self.assertLogs('popper') as test_logger:
+
+            result = runner.invoke(run.cli, ['-f', file_loc])
+            self.assertEqual(result.exit_code, 0)
+
+        self.assertTrue(len(test_logger.output))
```
Comment (ivotron) [inline cli/test/test_scaffold.py:29]: this should be indented within the `with` statement, as the `test_logger` variable is only valid in this scope

Comment (ivotron) [reply cli/test/test_scaffold.py:29]: we can also inspect the content of the output to ensure that it ran as it should

Comment (vipulchhabra99) [reply cli/test/test_scaffold.py:29]: Can I inspect the number of steps it ran because the other output is very lengthy?

Comment (ivotron) [reply cli/test/test_scaffold.py]: yeah, that works fine. we only need to solve the pep8 issues that might arise

Comment (vipulchhabra99) [reply cli/test/test_scaffold.py]: Updated.

**Comment by ivotron** [comment]
thanks a lot @vipulchhabra99 !

**Pull request merged**